src/openrct2/ride/coaster/GigaCoaster.cpp

```diff
@@ -9380,6 +9380,31 @@ static void GigaRCTrackBooster(
     PaintUtilSetGeneralSupportHeight(session, height + 32, 0x20);
 }
 
+static void GigaRCTrackPoweredLift(
+    PaintSession& session, const Ride& ride, uint8_t trackSequence, uint8_t direction, int32_t height,
+    const TrackElement& trackElement)
+{
+    PaintAddImageAsParentRotated(
+        session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(SPR_G2_GIGA_RC_POWERED_LIFT_0 + direction),
+        { 0, 0, height }, { 32, 20, 3 }, { 0, 6, height });
```

Comment:
Use BoundBoxXYZ

Reply:
I am looking into what you are trying to tell me but I cannot find any reference of BoundBoxXYZ currently existing in the track drawing code. So if you would like to have this changed around I think that'd be better for a separate refactor.

Reply:
In Paint.h:
```
PaintStruct* PaintAddImageAsParent(
    PaintSession& session, const ImageId image_id, const CoordsXYZ& offset, const CoordsXYZ& boundBoxSize);
PaintStruct* PaintAddImageAsParent(
    PaintSession& session, const ImageId image_id, const CoordsXYZ& offset, const CoordsXYZ& boundBoxSize,
    const CoordsXYZ& boundBoxOffset);
PaintStruct* PaintAddImageAsParent(
    PaintSession& session, const ImageId image_id, const CoordsXYZ& offset, const BoundBoxXYZ& boundBox);
[[nodiscard]] PaintStruct* PaintAddImageAsOrphan(
    PaintSession& session, const ImageId image_id, const CoordsXYZ& offset, const BoundBoxXYZ& boundBox);
PaintStruct* PaintAddImageAsChild(
    PaintSession& session, const ImageId image_id, const CoordsXYZ& offset, const BoundBoxXYZ& boundBox);

PaintStruct* PaintAddImageAsChildRotated(
    PaintSession& session, const uint8_t direction, const ImageId image_id, const CoordsXYZ& offset,
    const CoordsXYZ& boundBoxSize, const CoordsXYZ& boundBoxOffset);
PaintStruct* PaintAddImageAsParentRotated(
    PaintSession& session, const uint8_t direction, const ImageId image_id, const CoordsXYZ& offset,
    const CoordsXYZ& boundBoxSize);
PaintStruct* PaintAddImageAsParentRotated(
    PaintSession& session, const uint8_t direction, const ImageId imageId, const CoordsXYZ& offset,
    const CoordsXYZ& boundBoxSize, const CoordsXYZ& boundBoxOffset);
```
There is no BoundBoxXYZ overload for PaintAddImageAsParentRotated currently but you should create one and use it.

Reply:
This is kind beyond my understanding of the code.

Reply:
Right now, you use a CoordsXYZ for the offset, a CoordsXYZ for the boundbox offset and a CoordsXYZ for the boundbox size. I suggest you create a function signature that takes a CoordsXYZ for the offset and a BoundBoxXYZ for the boundbox and use it.

Reply:
Okay, but that's changing something around from the track element paint code. Wouldn't that be smarter for a separate PR?

Reply:
I agree, this can be addressed in a separate PR and streamline all of its uses afterwards.

Reply:
It has now become necessary to modify due to https://github.com/OpenRCT2/OpenRCT2/pull/19390 having been merged.

Reply:
```suggestion
        { 0, 0, height }, { { 0, 6, height }, { 32, 20, 3 } });
```